### PR TITLE
feat: show version tag in process instances table

### LINKED ISSUE #21327: Add version tag column to process instances list
### User Story

```Gherkin
Scenario: As an Operate user
    Given I am in process instances list view
    Then I can see the version tags in a column of the instances list
```
### Designs
https://www.figma.com/design/7lJFTkWKfhQUfM6nfil4Bh/Version-tag?node-id=189-29486&t=UgKu4aC2F4yljtHT-1

### Acceptance Criteria

When there is at least one version tag existing for any process definition: 
- [x] A new column "Tags" is visible in the process instances list
- [x] It shows the related version tag for each process definition

When there is no version tag for any process definition:
- [x]  Hide the "Tags" column

When a single process version is selected:
- [x] **Show** the "Tags" column when the selected version **has a version tag**
- [x] **Hide** the "Tags" column when the selected version **has no version tag**

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered and added to the product-hub issue risk assessment
- [ ] Cross-team dependencies have been considered

Optional:
- [ ] Design input has been collected by the assignee
### :point_right: Handover Dev to QA 

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Versions to validate:
> Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released)

### PR DESCRIPTION
**Pull request by pedesen**
## Description

- extend processes store with computed values for a flat list of process versions and for an indicator if there are any version tags.
- add a method to the processes store which returns the version tag for a specific process id
- add version tag column to instances list, if there are any
- show version tag for each process instance (if any)

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #21327

### CODE REVIEW

**Review by vitorwtavares — APPROVED** [review]

**Comment by pedesen** [comment]
@gastonpillet01: Please review using this image: `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-21327-version-tag-instances-table`.

Please note that in the test data "Complex Process" is the only process which has a version tag named "MyVersionTag".

**Review by gastonpillet01 — COMMENTED** [review]
I see:
- that Version tag column is always present not only for Complex process, if I select others process that hasn't version tag defined, column is being displayed with empty -- value
image

- in the designs it it was defined that if version tag is not defined Operate UI will not show Version tag column
image

In that way it will be consistent with the header behaviour about not showing version tag if it isn't defined.

**Comment by pedesen** [comment]
@gastonpillet01 Valid points, I will fix them.

File: operate/client/src/App/Processes/ListView/InstancesTable/index.tsx [hunk 1]
```diff
@@ -47,6 +47,13 @@ const InstancesTable: React.FC = observer(() => {
     state: {status, filteredProcessInstancesCount, processInstances},
   } = processInstancesStore;
 
+  /**
+   * Is true if at least one process instances from the store has a version tag.
+   */
+  const hasVersionTags = processInstances.some(({processId}) => {
+    return processesStore.getVersionTag(processId);
+  });
```
Comment (pedesen) [inline operate/client/src/App/Processes/ListView/InstancesTable/index.tsx:55]: Now I'm checking here if there is at least one process instance in the store with a related version. If there is none found, the version tag column header won't be shown.

**Comment by pedesen** [comment]
Hey @vitorwtavares could you please review an [additional commit](https://github.com/camunda/camunda/pull/21568/commits/4aeb8f2694e93f5a4ab9782db0997dc28d83c353) I added? based on Gaston's feedback I changed the behavior of determining whether the version tag column header is shown or not.

**Review by vitorwtavares — APPROVED** [review]

**Comment by pedesen** [comment]
@gastonpillet01 I updated the behavior. Now the version tag column is only shown if there is actually at least one process instance with a version tag in the list.

**Review by gastonpillet01 — APPROVED** [review]
Version tag is correctly presented in process instances table.

**Pull request merged**